### DISCUSSION TITLE
No internet response

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#164] [Course Qeustions]
I am working through the videos "Internet Server" -> "Creating a Server".
The wifi connects ok and then the server starts and gets and IP address. 
The PC is connected to same network as the ESP ("DrayTek"), When I run IP scanner, I see the esp device listed on IP 192.168.1.25 
However, when I open a browser and enter IP 192.168.1.25, the page does not load.

This site can’t be reached192.168.1.25 refused to connect. I also went to the MDNS video and set up the service and tried to navigate to http://my-esp32.local but that did not work either.

I can ping the esp IP address on PC and it is found ok.
C:\Users\Owner>ping 192.168.1.25

Pinging 192.168.1.25 with 32 bytes of data:
Reply from 192.168.1.25: bytes=32 time=146ms TTL=255
Reply from 192.168.1.25: bytes=32 time=232ms TTL=255
Reply from 192.168.1.25: bytes=32 time=45ms TTL=255
Reply from 192.168.1.25: bytes=32 time=57ms TTL=255

Ping statistics for 192.168.1.25:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 45ms, Maximum = 232ms, Average = 120ms

Here is output from monitor while running the code.

I (767) WIFI: Connecting...
I (797) wifi:new:<6,1>, old:<1,0>, ap:<255,255>, sta:<6,1>, prof:1
I (797) wifi:state: init -> auth (b0)
I (897) wifi:state: auth -> assoc (0)
I (967) wifi:state: assoc -> run (10)
W (987) wifi:<ba-add>idx:0 (ifx:0, 16:49:bc:42:f6:a0), tid:5, ssn:0, winSize:64
I (1027) wifi:connected with DrayTek, aid = 5, channel 6, 40U, bssid = 16:49:bc:42:f6:a0
I (1027) wifi:security: WPA2-PSK, phy: bgn, rssi: -83
I (1037) wifi:pm start, type: 1

I (1037) WIFI: Connected
W (1097) wifi:<ba-add>idx:1 (ifx:0, 16:49:bc:42:f6:a0), tid:0, ssn:4, winSize:64
I (1107) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (2057) WIFI: Got IP
I (2057) esp_netif_handlers: sta ip: 192.168.1.25, mask: 255.255.255.0, gw: 192.168.1.1

![IP Scan](https://user-images.githubusercontent.com/41096656/185247282-62fd871b-6369-4f97-a985-41c5efaf6345.png)

**Comment by orbitcoms** [comment]
Further to my post.
I swapped over to AP and used mdms and I get the "hello world" response ok (apart from error regarding "Favicon" if I view the developer page in chrome)

Still unable to get to it using the sta mode.

**Comment by Mair** [comment]
Are you sure the endpoints are being registered?
Perhaps add some logging to ensure it

**Reply by Mair** [reply]
Also might be worth download my code and changing the STA and password to see if its different

**Comment by orbitcoms** [comment]
Would the mode of connection (AP or STA) be affected by end point registration? ie. If there was an endpoint registered, would you expect it might work with AP but not STA method of connection?

**Comment by orbitcoms** [comment]
I'm a bit confused about accessing the ESP using a browser connected to the internet using STA. I understand that as an STYA it will connect as a station to my wifi network but when I open my browser, I am on a public network that does not have access through our firewall back into my organisations network. I thought you would need some www dns record to access threough browser?
I remember a while ago you used nxp http-server to be able to access it locally but this time you just seemed to use a browser. Am I missing something?  I am using Visual Studio 2022 with Visual GDB for development. Would I be better to install a local http server for development? What would you recommend?

**Comment by Mair** [comment]
Hi David,
As an STA on your local Wi-Fi. the ESP32 can reach out to the internet. If your pc is on the same Wi-Fi / network then the pc will be able to communicate to the ESP32.
If the pc is not on the same network. then you need to allow your ESP32 to be accessible to the internet. This can be done using port forwarding to your public IP
if your public IO changes you can can either get a static IP or you dynamic DNS
I probably need to talk to you about this in much more details if you want to have a chat
